django/chat/utils.py

```diff
@@ -327,53 +327,111 @@ def summarize_long_text(
-                "* The summary must not be longer than the input text.\n\n"
-                "Please rewrite the following document."
-                "\n\n Document: {docs}"
-            ),
-            "fr": (
-                "Réécrivez le texte (en anglais) sous forme de résumé détaillé, en utilisant plusieurs paragraphes si nécessaire. (Si la saisie est courte, affichez 1 seul paragraphe)\n\n"
-                "Quelques règles à suivre :\n"
-                '* Réécrivez simplement ; ne dites pas "Ce document concerne..." etc. Incluez *tous* les détails importants.\n'
-                "* Il n'y a pas de limite de longueur : soyez aussi détaillé que possible. Cependant, **n'extrapolez pas** sur le texte. Le résumé doit être factuel et ne pas introduire de nouvelles idées.\n"
-                "* Le résumé ne doit pas être plus long que le texte saisi.\n\n"
-                "Veuillez réécrire le document suivant."
-                "\n\n Document: {docs}"
-            ),
+            "en": """<document>
+{docs}
+</document>
+<instruction>
+Rewrite the text (in English) as a detailed summary, using multiple paragraphs if necessary. (If the input is short, output 1 paragraph only)
+
+Some rules to follow:
+* Simply rewrite; do not say "This document is about..." etc. Include *all* important details.
+* There is no length limit - be as detailed as possible.
+* **Never extrapolate** on the text. The summary must be factual and not introduce any new ideas.
+* If document is short, just output the document verbatim.
+</instruction>
+Detailed summary:
+""",
+            "fr": """<document>
+{docs}
+</document>
+<instruction>
+Réécrivez le texte (en anglais) sous forme de résumé détaillé, en utilisant plusieurs paragraphes si nécessaire. (Si la saisie est courte, affichez 1 seul paragraphe)
+
+Quelques règles à suivre :
+* Réécrivez simplement ; ne dites pas "Ce document concerne..." etc. Incluez *tous* les détails importants.
+* Il n'y a pas de limite de longueur : soyez aussi détaillé que possible.
+* **Ne faites jamais d'extrapolation** sur le texte. Le résumé doit être factuel et ne doit pas introduire de nouvelles idées.
+* Si le document est court, affichez-le tel quel.
+</instruction>
+Résumé détaillé :
+""",
         },
     }
 
     if custom_prompt and "{docs}" in custom_prompt:
         length_prompt_template = custom_prompt
     elif custom_prompt:
         length_prompt_template = (
-            custom_prompt
-            + "\n\n"
-            + _("The original document is below, enclosed in triple quotes:")
-            + "\n'''\n{docs}\n'''"
+            """
+<document>
+{docs}
+</document>
+<instruction>
+"""
+            + f"{custom_prompt}\n</instruction>"
         )
     else:
         length_prompt_template = length_prompts[length][target_language]
+        if gender_neutral:
```

Comment:
An interesting thing I noticed from playing around with elephants.msg and some small .txt files I made: including the gender-neutral instructions tends to dilute the "restate the document verbatim" instruction for short docs. You'll see "elephants are wonderful" when the gender-neutral instructions aren't included, and "the document celebrates elephants for being wonderful" when they are. Ditto for single-sentence .txt docs below a certain length.

Reply:
Hmm, I worried this would be the case. It might be better to just include the gender neutral instruction all the time, and PRIOR to the "repeat verbatim if short" instruction.
Unfortunately, the more instructions you add, the less it cares about any of them.